pennylane/transforms/transpile.py

```diff
@@ -0,0 +1,212 @@
+from pennylane.ops.qubit import SWAP
+from pennylane.operation import Tensor
+from pennylane.ops import __all__ as all_ops
+from pennylane.tape import QuantumTape
+from pennylane.transforms import qfunc_transform
+from pennylane.tape import stop_recording
+from pennylane.wires import Wires
+
+
+@qfunc_transform
+def transpile(tape: QuantumTape, coupling_map: Union[List, nx.Graph]):
+    """Transpile a circuit according to a desired coupling map
+
+    .. warning::
+
+        This transform does not yet support measurements of Hamiltonians or tensor products of observables. If a circuit
+        is passed which contains these types of measurements, a NotImplementedError will be raised.
+
+    Args:
+        tape (function): A quantum function.
+        coupling_map (list[tuple(int, int)] or nx.Graph): Either a list of tuples(int, int) or an instance of
+            networkx.Graph specifying the couplings between different qubits.
+
+    Returns:
+        function: the transformed quantum function
+
+    **Example**
+
+    Consider the following example circuit
+
+    .. code-block:: python
+
+        def circuit():
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[2, 3])
+            qml.CNOT(wires=[1, 3])
+            qml.CNOT(wires=[1, 2])
+            qml.CNOT(wires=[2, 3])
+            qml.CNOT(wires=[0, 3])
+            return qml.probs(wires=[0, 1, 2, 3])
+
+    which, before transpiling it looks like this:
+
+    .. code-block:: text
+
+        0: ──╭C──────────────╭C──╭┤ Probs
+        1: ──╰X──╭C──╭C──────│───├┤ Probs
+        2: ──╭C──│───╰X──╭C──│───├┤ Probs
+        3: ──╰X──╰X──────╰X──╰X──╰┤ Probs
+
+    Suppose we have a device which has connectivity constraints according to the graph:
+
+    .. code-block:: text
+
+        0 --- 1
+        |     |
+        2 --- 3
+
+    We encode this in a coupling map as a list of the edges which are present in the graph, and then pass this, together
+    with the circuit, to the transpile function to get a circuit which can be executed for the specified coupling map:
+
+    >>> dev = qml.device('default.qubit', wires=[0, 1, 2, 3])
+    >>> transpiled_circuit = qml.transforms.transpile(coupling_map=[(0, 1), (1, 3), (3, 2), (2, 0)])(circuit)
+    >>> transpiled_qnode = qml.QNode(circuit, dev)
+    >>> print(qml.draw(transpiled_qnode))
+    0: ──╭C─────────────────────╭C──╭┤ Probs
+    1: ──╰X──╭C─────────╭C──────│───├┤ Probs
+    2: ──╭C──│───╭SWAP──│───╭X──╰X──├┤ Probs
+    3: ──╰X──╰X──╰SWAP──╰X──╰C──────╰┤ Probs
+
+    A swap gate has been applied to wires 2 and 3, and the remaining gates have been adapted accordingly
+
+    """
+    # init connectivity graph
+    coupling_graph = (
+        nx.Graph(coupling_map) if not isinstance(coupling_map, nx.Graph) else coupling_map
+    )
+
+    # make sure every wire is present in coupling map
+    if any(wire not in coupling_graph.nodes for wire in tape.wires):
+        wires = [w for w in tape.wires]
```

Comment:
```suggestion
        wires = tape.wires.tolist()
```

Reply:
Let me commit this small change to see if that kickstarts the tests 🤔

Reply:
ah cool, this did the trick^^